source/iface/usm_basic_concept.rst

```diff
@@ -2,288 +2,291 @@
+  * - ``shared``
+    - Allocations in shared memory that are accessible by both host and device.
+
+USM is an optional feature which may not be supported by all devices, and
+devices that support USM may not support all types of USM allocation.
+
+A SYCL application can use the ``sycl::device::has()``
+function to determine the level of USM support for a device
+(See :ref:`device-aspects`).
+
+.. seealso:: |SYCL_SPEC_USM_KINDS|
+
+Rules of USM allocations
+------------------------
+
+Each USM allocation has an associated SYCL context, and any access to
+that memory must use the same context.
+Specifically, any SYCL kernel function that dereferences a pointer to
+a USM allocation must be submitted to a :ref:`queue` that was constructed
+with the same context that was used to allocate that memory.
+The explicit memory operation commands that take USM pointers have a
+similar restriction.
+
+.. warning::
+
+  Violations of these requirements result in undefined behavior.
+
+.. note::
+
+  There are no similar restrictions for dereferencing a USM pointer in a
+  host task. This is legal regardless of which queue the host task
+  was submitted to so long as the USM pointer is accessible on the host.
+
+.. warning::
+
+  Each type of USM allocation has different rules for where that memory
+  is accessible. Attempting to dereference a USM pointer on the host or
+  on a device in violation of these rules results in undefined behavior.
+
+  Passing a USM pointer to one of the explicit memory functions where
+  the pointer is not accessible to the device generally results in
+  undefined behavior.
+
+
+Host allocations
+----------------
+
+Host allocations allow devices to directly read and write host
+memory inside of a kernel.
+
+Host allocations must also be obtained using SYCL routines
+instead of system allocation routines. While a device may remotely
+read and write a host allocation, the allocation does not migrate
+to the device - it remains in host memory.
+
+.. warning::
+
+  Users should take care to properly synchronize access to
+  host allocations between host execution and kernels.
+
+The total size of host allocations will be limited by the amount
+of pinnable-memory on the host on most systems.
+
+Support for host allocations on a specific device can be queried
+through ``sycl::aspect::usm_host_allocations``.
+
+Support for atomic modification of host allocations on a specific
+device can be queried through ``sycl::aspect::usm_atomic_host_allocations``.
+
+
+Device allocations
+------------------
+
+Device allocations are used for explicitly managing device memory.
+
+Device allocations are obtained through SYCL device USM allocation
+routines instead of system allocation routines like ``std::malloc``
+or C++ ``new``.
+
+Device allocations are not accessible on the host, but the pointer
+values remain consistent on account of :ref:`unified_addressing`.
+
+With device allocations data directly allocated in the device memory
```

Comment:
```suggestion
With device allocations data is directly allocated in the device memory
```

Reply:
Added in the https://github.com/KhronosGroup/SYCL_Reference/pull/82/commits/9a3cc5abdf86305f9a1cdd676563bf6fb64695f2.